staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go

```diff
@@ -291,6 +320,33 @@ func (c *threadSafeMap) Replace(items map[string]interface{}, resourceVersion st
 	}
 }
 
+func (c *threadSafeMap) rvFromObject(obj interface{}) (rv string, err error) {
+	meta, err := meta.Accessor(obj)
+	if err != nil {
+		return
+	}
+	rv = meta.GetResourceVersion()
+	_, err = resourceversion.CompareResourceVersion(rv, rv)
+	if err != nil {
+		return "", err
+	}
```

Comment:
do we need to do this in rvFromObject or only in updateRV? ideally we wouldn't do both